src/pages/NewPost/ErrorAlert.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+
+export default function AlertDialog({ isAlertOpen, setAlert, errorKey }) {
+  const handleClose = () => {
+    setAlert({ open: false, key: '' });
+  };
+
+  return (
+    <div>
+      <Dialog
+        open={isAlertOpen}
+        onClose={handleClose}
+        aria-labelledby="Error"
+        aria-describedby="There is something wrong."
+      >
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            {`You need to add ${
+              errorKey === 'categories' || errorKey === 'something' ? '' : 'a '
```

Comment:
Really like the error handling. The only thing I would suggest is separate messages to handle zero categories checked and too many categories checked.

Reply:
Done, error messages and handling largely improved to deal with much more scenarios